### PR TITLE
Add support for title attribute specification in TOC.md files

### PR DESCRIPTION
**Pull request by DuncanmaMSFT**
Using the valid markdown syntax for the title attribute in a link to add a displayText attribute to resulting TOC JSON file.

### CODE REVIEW

File: test/Microsoft.DocAsCode.Build.TableOfContents.Tests/MarkdownTocReaderTest.cs [hunk 1]
```diff
@@ -17,7 +17,7 @@ public void TestTocMdReader()
             var toc = MarkdownTocReader.LoadToc(@"
 # [Article1](article1.md)
 ## Container1 ##
-### [Article2](article2.md) ## 
+### [Article2](article2.md ""Article 2"") ##
```
Comment (superyyrrzz) [inline test/Microsoft.DocAsCode.Build.TableOfContents.Tests/MarkdownTocReaderTest.cs]: What if there's a file named `article2.md "Article 2"`? What about using query string like `(article2.md?displayText="Article 2")`? @vwxyzh

File: src/Microsoft.DocAsCode.DataContracts.Common/TocItemViewModel.cs [hunk 2]
```diff
@@ -23,6 +23,11 @@ public class TocItemViewModel
         [JsonProperty(Constants.PropertyName.Name)]
         public string Name { get; set; }
 
+        [YamlMember(Alias = Constants.PropertyName.DisplayName)]
+        [JsonProperty(Constants.PropertyName.DisplayName)]
+        public string DisplayName { get; set; }
+
+
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.DataContracts.Common/TocItemViewModel.cs]: remove extra empty line

File: src/Microsoft.DocAsCode.DataContracts.Common/TocItemViewModel.cs [hunk 3]
```diff
@@ -23,6 +23,11 @@ public class TocItemViewModel
         [JsonProperty(Constants.PropertyName.Name)]
         public string Name { get; set; }
 
+        [YamlMember(Alias = Constants.PropertyName.DisplayName)]
+        [JsonProperty(Constants.PropertyName.DisplayName)]
+        public string DisplayName { get; set; }
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.DataContracts.Common/TocItemViewModel.cs]: >DisplayName [](start = 22, length = 11)

Title instead?

**Comment by chenkennt** [comment]
@DuncanmaMSFT to unblock you, I will the PR, please take a look at the comments later.

**Pull request merged**